### ISSUE TITLE
Panic when stopping TunAdapter

### ISSUE THREAD

**Issue by Arceliar** [yggdrasil-network/yggdrasil-go#521]
Not sure if it is related, but let it be here:
```
2019/08/22 08:12:38 Stopping...
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x8 pc=0x5ef281]

goroutine 1 [running]:
github.com/yggdrasil-network/yggdrasil-go/src/tuntap.(*TunAdapter).Stop(...)
	E:/_Projects/_Test/yggdrasil-go/src/tuntap/tun.go:176
main.(*node).shutdown(0xc0000af500)
	E:/_Projects/_Test/yggdrasil-go/cmd/yggdrasil/main.go:297 +0xc1
main.main()
	E:/_Projects/_Test/yggdrasil-go/cmd/yggdrasil/main.go:291 +0xe1d
```

_Originally posted by @Vort in https://github.com/yggdrasil-network/yggdrasil-go/pull/517#issuecomment-523751011_

**Comment by Arceliar** [comment]
https://github.com/yggdrasil-network/yggdrasil-go/blob/562a7d1f19f254e9d078aef44a78f7d1398bc4e5/src/tuntap/tun.go#L172-L178

It looks like the crash happens on `tun.iface.Close()`, so my guess is that iface is `nil`. Not sure why. Maybe it's running on Windows and tuntap is disabled in the config?

**Comment by Vort** [comment]
I guess I know what happened.
Sometimes start of TUN/TAP fails because of several bugs ("Failed to find the name of interface", "The system cannot find the file specified", etc.).
After I see that start is failed, I restart program. And then I get this error.

**Comment by Arceliar** [comment]
Should be fixed in #526, to the extent that it won't panic if the tap wasn't set up correctly. The tap not being set up correctly is another issue...

**Comment by Vort** [comment]
Did your solution was designed to handle `IfName: none` case?
I see this error with `0.3.8-0067` in such case:
```
2019/09/02 13:00:24 Stopping...
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x8 pc=0x5ebd50]

goroutine 1132 [running]:
github.com/yggdrasil-network/yggdrasil-go/src/tuntap.(*TunAdapter)._stop(...)
        /home/circleci/project/src/tuntap/tun.go:194
github.com/yggdrasil-network/yggdrasil-go/src/tuntap.(*TunAdapter).Stop.func1()
        /home/circleci/project/src/tuntap/tun.go:185 +0x40
github.com/Arceliar/phony.Block.func1()
        /go/pkg/mod/github.com/!arceliar/phony@v0.0.0-20190831050304-94a6d3da9ba
4/actor.go:83 +0x36
github.com/Arceliar/phony.(*Inbox).run(0xc000097dd0)
        /go/pkg/mod/github.com/!arceliar/phony@v0.0.0-20190831050304-94a6d3da9ba
4/actor.go:92 +0xdf
created by github.com/Arceliar/phony.(*Inbox).enqueue
        /go/pkg/mod/github.com/!arceliar/phony@v0.0.0-20190831050304-94a6d3da9ba
4/actor.go:60 +0xda
```
